aidial_adapter_vertexai/chat/tools.py

```diff
@@ -137,9 +141,9 @@ def from_request(cls, request: AzureChatCompletionRequest) -> Self:
 
         return cls(functions=selected, required=required, tool_ids=tool_ids)
 
-    def to_gemini_tools(self) -> List[GeminiTool] | None:
+    def to_gemini_tools(self) -> List[GeminiTool]:
         if not self.functions:
-            return None
+            return []
```

Comment:
This fix leads to a slight regression in the request sent to the upstream model:
previously we sent `tools=None` iff there were no tools/functions, now we **always** send `tools=[]`.